macros/compat/src/main/scala-2.10/enumeratum/ContextUtils.scala

```diff
@@ -37,4 +37,10 @@ object ContextUtils {
   def constructorName(c: Context): c.universe.TermName = {
     c.universe.nme.CONSTRUCTOR
   }
+
+  def evalTree[ValueType](c: Context)(expr: c.universe.Tree): ValueType = {
+    @SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements")) // false positive bug in WartRemover?
+    val evaluated = c.eval(c.Expr[ValueType](expr))
```

Comment:
Scala 2.10 doesn't support typecheck and untypecheck but it seems to just work anyway.

Reply:
Yeah, the Scala macros API changed quite a bit between 2.10 and 2.11, hence the ContextUtils shim. Thanks for doing the legwork to get it working!!